src/EditorFeatures/Core.Wpf/Suggestions/PreviewChanges/PreviewChangesSuggestedAction.cs

```diff
@@ -31,21 +29,8 @@ private sealed partial class PreviewChangesSuggestedAction : SuggestedAction
             {
             }
 
-            public static async Task<SuggestedAction> CreateAsync(
-                SuggestedActionWithNestedFlavors suggestedAction, CancellationToken cancellationToken)
+            public static SuggestedAction Create(SuggestedActionWithNestedFlavors suggestedAction)
             {
-                var previewResult = await suggestedAction.GetPreviewResultAsync(cancellationToken).ConfigureAwait(true);
-                if (previewResult == null)
-                {
-                    return null;
-                }
-
-                var changeSummary = previewResult.ChangeSummary;
-                if (changeSummary == null)
-                {
-                    return null;
-                }
```

Comment:
so, slightly concerned about old vs new semantics.  in the old semantics, we wouldn't make the PreviewChanges action.  Now we do... but it may not know it has no contents till later?  Trying to figure out what this will appear to be like for the user.  Can you clarify when this happens?

Reply:
I'm not sure what happens in this case.

Reply:
I don’t have context on this either, I’ll try to dig around

Reply:
looks like we still need info on this.

Reply:
@mavasani We are paying a _heavy_ penalty for the current behavior in CI. Can you look into whether this change can be completed?

Reply:
@sharwell So, if I am understanding this correctly, this PR is only affecting how we create preview changes code action for the nested actions in a group. I am wondering if the original issue that led you to investigate this is fixed by https://github.com/dotnet/roslyn/pull/68859, which caused nested actions and their preview computation to intermittently not complete. I'll investigate the behavior changes with this PR, but we may want to double check if the current behavior in CI is resolved with #68859.

Reply:
@sharwell I am hitting the following assert when clicking the Preview Changes hyperlink in the preview window on your branch:

```
---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------
When created thread id  : 1

When created thread name: 

Current thread id       : 68

Current thread name     : 



   at Microsoft.CodeAnalysis.Editor.Shared.Utilities.ForegroundThreadAffinitizedObject.AssertIsForeground() in C:\roslyn\src\EditorFeatures\Core\Shared\Utilities\ForegroundThreadAffinitizedObject.cs:line 48

   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.<GetPreviewResultAsync>d__26.MoveNext() in C:\roslyn\src\EditorFeatures\Core.Wpf\Suggestions\SuggestedActions\SuggestedAction.cs:line 224

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.GetPreviewResultAsync(CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionWithNestedFlavors.PreviewChangesSuggestedAction.PreviewChangesCodeAction.<GetOperationsCoreAsync>d__6.MoveNext() in C:\roslyn\src\EditorFeatures\Core.Wpf\Suggestions\PreviewChanges\PreviewChangesCodeAction.cs:line 44

   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine)

   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedActionWithNestedFlavors.PreviewChangesSuggestedAction.PreviewChangesCodeAction.GetOperationsCoreAsync(Solution originalSolution, IProgressTracker progressTracker, CancellationToken cancellationToken)

   at Microsoft.CodeAnalysis.CodeActions.CodeAction.GetOperationsAsync(Solution originalSolution, IProgressTracker progressTracker, CancellationToken cancellationToken) in C:\roslyn\src\Workspaces\Core\Portable\CodeActions\CodeAction.cs:line 144

   at Microsoft.CodeAnalysis.Editor.Implementation.Suggestions.SuggestedAction.<>c__DisplayClass13_0.<GetOperationsAsync>b__0() in C:\roslyn\src\EditorFeatures\Core.Wpf\Suggestions\SuggestedActions\SuggestedAction.cs:line 80

   at System.Threading.Tasks.Task`1.InnerInvoke()

   at System.Threading.Tasks.Task.Execute()

   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)

   at System.Threading.Tasks.Task.ExecuteWithThreadLocal(Task& currentTaskSlot)

   at System.Threading.Tasks.Task.ExecuteEntry(Boolean bPreventDoubleExecution)

   at System.Threading.ThreadPoolWorkQueue.Dispatch()


---------------------------
Abort   Retry   Ignore   
---------------------------
```

Reply:
> So, if I am understanding this correctly, this PR is only affecting how we create preview changes code action for the nested actions in a group

The issue I'm resolving is the preview operations are getting fully realized by the unit testing framework just because we asked for the _name_. In other words, as part of figuring out what needs to show in the context menu, we calculate the content of a fly-out that isn't needed unless a user selects one of those menu items.

Reply:
@sharwell The code being changed in this PR doesn’t affect the flyout. It only affects whether or not Preview Changes hyperlink shows up and the code action to be applied when clicked on that link.

Reply:
> I am hitting the following assert when clicking the Preview Changes hyperlink in the preview window on your branch

I pushed a fix for this. Async methods are expected to switch to the expected thread, and not make assertions about the caller.